x-pack/plugins/ml/server/models/job_service/jobs.ts

```diff
@@ -257,6 +259,32 @@ export function jobsProvider(client: IScopedClusterClient, mlClient: MlClient) {
     return { jobs, jobsMap };
   }
 
+  async function getJobForCloning(jobId: string) {
+    const datafeedId = await getDatafeedIdByJobId(jobId);
+
+    const [{ body: jobResults }, { body: datafeedResults }] = await Promise.all([
+      mlClient.getJobs<MlJobsResponse>({ job_id: jobId, exclude_generated: true }),
+      mlClient.getDatafeeds<MlDatafeedsResponse>({
+        datafeed_id: datafeedId,
+        exclude_generated: true,
+      }),
+    ]);
+
+    let datafeed;
+    if (Array.isArray(datafeedResults?.datafeeds) && datafeedResults.datafeeds.length === 1) {
+      datafeed = datafeedResults?.datafeeds[0];
+    }
+
+    // create jobs objects containing job stats, datafeeds, datafeed stats and calendars
+    if (jobResults && jobResults.jobs) {
+      const job = jobResults.jobs.find((j) => j.job_id === jobId);
+      if (job && datafeed) {
+        return { job, datafeed };
```

Comment:
there is a chance that the datafeed for a job can be missing.
I think this should always return an object, but with `job` and `datafeed` possibly being `undefined`

Reply:
Updated here https://github.com/elastic/kibana/pull/88898/commits/6ea550868c19746c3fe817c19195cbb492f36eba